flask_server.py

```diff
@@ -65,19 +96,24 @@ def settings():
     settings = get_settings()    # Gets the settings
     cookie_path = settings["cookie_path"] if "cookie_path" in settings else "token.txt"
     folder_path = settings["folder_path"]  if "folder_path" in settings else "ig_media"
+    updated_settings = {}
 
     if request.method == "POST":
-        updated_settings = {}
-
         for setting in request.form:
             if len(request.form[setting]) > 0:
                 updated_settings[setting] = request.form[setting]
         save_settings(updated_settings)
+    return render_template("settings.html", folder_path = updated_settings["folder_path"] if "folder_path" in updated_settings else folder_path, cookie_path = updated_settings["cookie_path"] if "cookie_path" in updated_settings else cookie_path)
+
+@app.route("/gallery/", methods=['GET'])
```

Comment:
Create differents app/route

gallery/
gallery/{username}
gallery/{username}/{date}

Reply:
1st start search in base_folder
2st os.path.join start there